packages/build/tests/tracing/tests.js

```diff
@@ -11,6 +17,70 @@ test('Tracing set multi span attributes', async (t) => {
   t.is(baggage.getEntry('foo').value, 'bar')
 })
 
+const testMatrixBaggageFile = [
+  {
+    description: 'when baggageFilePath is blank',
+    input: {
+      baggageFilePath: '',
+      baggageFileContent: null,
+    },
+    expects: {
+      somefield: undefined,
+      foo: undefined,
+    },
+  },
+  {
+    description: 'when baggageFilePath is set but file is empty',
+    input: {
+      baggageFilePath: BAGGAGE_PATH,
+      baggageFileContent: '',
+    },
+    expects: {
+      somefield: undefined,
+      foo: undefined,
+    },
+  },
+  {
+    description: 'when baggageFilePath is set and has content',
+    input: {
+      baggageFilePath: BAGGAGE_PATH,
+      baggageFileContent: 'somefield=value,foo=bar',
+    },
+    expects: {
+      somefield: { value: 'value' },
+      foo: { value: 'bar' },
+    },
+  },
+]
+
+testMatrixBaggageFile.forEach((testCase) => {
+  test.serial(`Tracing baggage loading - ${testCase.description}`, async (t) => {
+    const { input, expects } = testCase
+    if (input.baggageFilePath.length > 0) {
+      await mkdir(FIXTURES_DIR, { recursive: true })
+      await writeFile(input.baggageFilePath, input.baggageFileContent)
+    }
+
+    const ctx = loadBaggageFromFile(input.baggageFilePath)
+    const baggage = propagation.getBaggage(ctx)
+
+    Object.entries(expects).forEach(([property, expected]) => {
+      if (input.baggageFilePath === '') {
+        t.is(baggage, undefined)
+      } else {
```

Comment:
We can probably do this outside of the entries iteration? And thereby just avoid having the expects altogether?